File: externals/Async-CTP/samples/VB/Windows/Phone/Concert/Finder/Concert/Finder/Converters/BooleanOpacityConverter.vb
Imports System
Imports System.Globalization
Imports System.Windows.Data

Namespace ConcertFinder.Converters

    ''' <summary>
    ''' Converter for boolean values.
    ''' </summary>
    Public Class BooleanOpacityConverter
        Implements IValueConverter

        ''' <summary>
        ''' Convert a boolean value to an opacity.
        ''' </summary>
        Public Function Convert(value As Object, targetType As [Type], parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim convertedValue = 1
            Dim originalValue = DirectCast(value, Boolean)
            convertedValue = If((originalValue = True), 1, 0)
            Return convertedValue
        End Function

        Public Function ConvertBack(value As Object, targetType As [Type], parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function

    End Class

End Namespace

File: externals/Async-CTP/samples/CS/Windows/Phone/Concert/Finder/Concert/Finder/Converters/BooleanOpacityConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace ConcertFinder.Converters
{
    /// <summary>
    /// Converter for boolean values.
    /// </summary>
    public class BooleanOpacityConverter : IValueConverter
    {
        /// <summary>
        /// Convert a boolean value to an opacity.
        /// </summary>
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var convertedValue = 1;

            var originalValue = (bool)value;
            convertedValue = (originalValue == true) ? 1 : 0;

            return convertedValue;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}
